client/src/components/UserProfileChangeEmail.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div class="row">
+    <div class="col-md-8 mb-3">
+      <BaseInput
+        id="email"
+        v-model="modifiedUserData.email"
+        :is-email="true"
+        label="E-Mail"
+        :is-disabled="emailFieldIsDisabled"
+      />
+    </div>
+  </div>
+  <div class="col-md-6">
+    <button
+      class="btn btn-primary"
+      :disabled="!saveButtonIsEnabled"
+      data-cy="email-change-btn"
+      @click="onSave"
+    >
+      Email ändern
+      <BaseSpinner v-if="showSpinner" />
+    </button>
+    <BaseError :error-message="errorMessage" class="my-2" />
+  </div>
+</template>
+<script setup>
+// TODO: How to test this?
+
+import ApiService from "@/services/ApiService.js";
+import { ref, computed } from "vue";
+import BaseSpinner from "./BaseSpinner.vue";
+import BaseError from "./BaseError.vue";
+import Swal from "sweetalert2";
+import { isEmail } from "../helper/utils";
+import useUserProfile from "@/composables/useUserProfile";
+
+const { modifiedUserData, emailHasChanged } = useUserProfile();
+
+// Page state
+const showSpinner = ref(false);
+const errorMessage = ref(null);
+const emailFieldIsDisabled = ref(false);
+
+const inidcateSuccess = () => {
+  Swal.fire({
+    icon: "success",
+    // TODO: Set color globally
```

Comment:
Wir sollten dann evtl. die Background Color noch in Abhänigkeit des Themes wählen

Reply:
habs auf die todo gepackt